plasma_framework/migrations/1041_setup_vault.js

```diff
@@ -0,0 +1,171 @@
+const sleep = (milliseconds) => {
+    return new Promise(resolve => setTimeout(resolve, milliseconds));
+};
+
+module.exports = async (
+    deployer,
+    _,
+    // eslint-disable-next-line no-unused-vars
+    [deployerAddress, maintainerAddress, authorityAddress],
+) => {
+    const vault = process.env.VAULT || false;
+    if (vault) {
+        console.log('Performing final setup with multisig maintainer');
+        const plasmaFramework = await PlasmaFramework.deployed();
+        const sha = childProcess.execSync('git rev-parse HEAD').toString().trim().substring(0, 7);
+        const ethDepositVerifier = await EthDepositVerifier.deployed();
+        const ethVault = await EthVault.deployed();
+        const erc20DepositVerifier = await Erc20DepositVerifier.deployed();
+        const erc20Vault = await Erc20Vault.deployed();
+        const MORE_VP = config.frameworks.protocols.moreVp;
+        const PAYMENT_TX_TYPE = config.registerKeys.txTypes.payment;
+        const FEE_TX_TYPE = config.registerKeys.txTypes.fee;
+        const paymentExitGame = await PaymentExitGame.deployed();
+        const feeExitGame = await FeeExitGame.deployed();
+        const buildDir = path.resolve(__dirname, '../../MultiSigWallet/build/multisig_instance');
+        const gnosisMultisigAddress = fs.readFileSync(buildDir, 'utf8');
+        const gnosisMultisigAbi = { constant: false,
+            inputs: [
+                { name: 'destination', type: 'address' },
+                { name: 'value', type: 'uint256' },
+                { name: 'data', type: 'bytes' },
+            ],
+            name: 'submitTransaction',
+            outputs: [{ name: 'transactionId', type: 'uint256' }],
+            payable: false,
+            type: 'function',
+            signature: '0xc6427474' };
+        // ethVault.setDepositVerifier
+        const setDepositVerifier = web3.eth.abi.encodeFunctionCall(ethVault.abi.find(o => o.name === 'setDepositVerifier'), [ethDepositVerifier.address]);
+        const gnosisSetDepositVerifier = web3.eth.abi.encodeFunctionCall(gnosisMultisigAbi, [ethVault.address, 0, setDepositVerifier]);
+        /* eslint-disable prefer-arrow-callback */
+        web3.eth.sendTransaction({ gas: 3000000, to: gnosisMultisigAddress, from: deployerAddress, data: gnosisSetDepositVerifier }, async function (error, transactonHash) {
+            console.log(`Submitted transaction with hash: ${transactonHash}`);
+            let transactionReceipt = null;
+            while (transactionReceipt == null) { // Waiting expectedBlockTime until the transaction is mined
+                /* eslint-disable no-await-in-loop */
+                transactionReceipt = await web3.eth.getTransactionReceipt(transactonHash);
+                /* eslint-disable no-await-in-loop */
+                await sleep(expectedBlockTime);
+            }
+            console.log(`Got the transaction receipt for ETH setDepositVerifier: ${transactionReceipt}`);
```

Comment:
looks like we wait for the receipt but it isn't logged at the end 🤔 Does anything silently fail?

Reply:
I'm not sure I understand. What isn't logged? Ah, do you think I should look at status from the transactionReceipt and abort if it's not true?

Reply:
Ohh the transactionReceipt body's aren't logged i think. Yeah I was also thinking about looking at the status and handling if a failed transaction, though not a necessity maybe but might be good to have

Reply:
Probably just logging out transaction receipts to show which failed or aborting

Reply:
I think I'll abort. Because we insert 8 transactions into the gnosis multisig, all of them need to be executed "in order" they were submitted. So it's easier to know where things broke and where to continue.